services/brig/src/Brig/API/Public.hs

```diff
@@ -1115,13 +1090,12 @@ customerExtensionCheckBlockedDomains email = do
         when (domain `elem` blockedDomains) $
           throwM $ customerExtensionBlockedDomain domain
 
-createConnectionH :: JSON ::: UserId ::: ConnId ::: JsonRequest Public.ConnectionRequest -> Handler Response
-createConnectionH (_ ::: self ::: conn ::: req) = do
-  cr <- parseJsonBody req
+createConnectionH :: UserId -> ConnId -> Public.ConnectionRequest -> Handler (Public.ResponseForExistedCreated Public.UserConnection)
+createConnectionH self conn cr = do
   rs <- API.createConnection self cr conn !>> connError
   return $ case rs of
-    ConnectionCreated c -> setStatus status201 $ json (c :: Public.UserConnection)
-    ConnectionExists c -> json (c :: Public.UserConnection)
+    ConnectionCreated c -> Public.Created201 c
```

Comment:
There's no need to have this conversion here. We can just use the `ResponseForExistedCreated` type as the return type of `createConnection`.

Reply:
Good point. Fixed.